include/accessor.h

```diff
@@ -199,47 +175,96 @@ class accessor<DataT, Dims, Mode, target::device> : public detail::accessor_base
+	    const buffer<DataT, Dims>& buff, handler& cgh, const Functor& rmfn, const detail::access_tag<Mode, TagModeNoInit, target::device> /* tag */)
+	    : accessor(ctor_internal_tag(), buff, cgh, rmfn) {}
+
+	template <typename Functor, access_mode TagMode>
+	CELERITY_INTERNAL_ACCESSOR_DEPRECATED_CTOR accessor(const buffer<DataT, Dims>& buff, handler& cgh, const Functor& rmfn,
+	    const detail::access_tag<TagMode, Mode, target::device> /* tag */, const property::no_init& /* no_init */)
+	    : accessor(ctor_internal_tag(), buff, cgh, rmfn) {}
+
+	template <typename Functor, access_mode TagMode, access_mode TagModeNoInit>
+	CELERITY_INTERNAL_ACCESSOR_DEPRECATED_CTOR accessor(const buffer<DataT, Dims>& buff, handler& cgh, const Functor& rmfn,
+	    const detail::access_tag<TagMode, TagModeNoInit, target::device> /* tag */, const property_list& /* prop_list */) {
+		static_assert(detail::constexpr_false<Functor>,
+		    "Currently it is not accepted to pass a property list to an accessor constructor. Please use the property celerity::no_init "
+		    "as a last argument in the constructor");
+	}
+
+	template <int D = Dims, std::enable_if_t<D == 0, int> = 0>
+	CELERITY_INTERNAL_ACCESSOR_DEPRECATED_CTOR accessor(const buffer<DataT, Dims>& buff, handler& cgh) : accessor(buff, cgh, access::all()) {}
 
 	template <access_mode TagModeNoInit, int D = Dims, std::enable_if_t<D == 0, int> = 0>
-	accessor(const buffer<DataT, Dims>& buff, handler& cgh, const detail::access_tag<Mode, TagModeNoInit, target::device> tag)
+	CELERITY_INTERNAL_ACCESSOR_DEPRECATED_CTOR accessor(
+	    const buffer<DataT, Dims>& buff, handler& cgh, const detail::access_tag<Mode, TagModeNoInit, target::device> tag)
 	    : accessor(buff, cgh, access::all(), tag) {}
 
 	template <access_mode TagMode, int D = Dims, std::enable_if_t<D == 0, int> = 0>
-	accessor(const buffer<DataT, Dims>& buff, handler& cgh, const detail::access_tag<TagMode, Mode, target::device> tag, const property::no_init& no_init)
+	CELERITY_INTERNAL_ACCESSOR_DEPRECATED_CTOR accessor(
+	    const buffer<DataT, Dims>& buff, handler& cgh, const detail::access_tag<TagMode, Mode, target::device> tag, const property::no_init& no_init)
 	    : accessor(buff, cgh, access::all(), tag, no_init) {}
 
 	template <access_mode TagMode, access_mode TagModeNoInit, int D = Dims, std::enable_if_t<D == 0, int> = 0>
-	accessor(
+	CELERITY_INTERNAL_ACCESSOR_DEPRECATED_CTOR accessor(
 	    const buffer<DataT, Dims>& buff, handler& cgh, const detail::access_tag<TagMode, TagModeNoInit, target::device> tag, const property_list& prop_list)
 	    : accessor(buff, cgh, access::all(), tag, prop_list) {}
 
 	// explicitly defaulted because we define operator=(value_type) for Dims == 0
-	accessor(const accessor&) = default;
 	accessor(accessor&&) noexcept = default;
-	accessor& operator=(const accessor&) = default;
 	accessor& operator=(accessor&&) noexcept = default;
 
+#if !defined(__SYCL_DEVICE_ONLY__) && !defined(SYCL_DEVICE_ONLY)
```

Comment:
I assume these relate to different SYCL backends? Might make sense to define a `CELERITY_DEVICE_ONLY` somewhere to make this more readable (and easier to adapt in case some SYCL implementation tries to use a third variant).

Reply:
Yes, `__SYCL_DEVICE_ONLY__` is defined by the standard, but hipSYCL for some reason always used the non-standard `SYCL_DEVICE_ONLY`. However, I just checked, and it seems that this has since been fixed, so we can just resort to the standard one. Or define `CELERITY_DEVICE_ONLY`. Both are fine with me!

Reply:
As long as all supported platforms agree on one thing I don't think we need our own.